source/common/upstream/health_checker_impl.cc

```diff
@@ -206,9 +207,17 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onTimeout() {
   client_->close();
 }
 
+Http::CodecClient::Type HttpHealthCheckerImpl::codecClientType(bool use_http2) {
+  if (use_http2) {
+    return Http::CodecClient::Type::HTTP2;
+  } else {
+    return Http::CodecClient::Type::HTTP1;
+  }
+}
+
 Http::CodecClient*
 ProdHttpHealthCheckerImpl::createCodecClient(Upstream::Host::CreateConnectionData& data) {
-  return new Http::CodecClientProd(Http::CodecClient::Type::HTTP1, std::move(data.connection_),
+  return new Http::CodecClientProd(codec_client_type_, std::move(data.connection_),
```

Comment:
From my recollection, this function (the prod version) is not actually covered in an any test. Do you mind checking the CI coverage report? I'm wondering since you already already in here if there is any way to get this covered?

Reply:
I think you're right, I'll try to add a test

Wasn't able to figure out how to access the coverage report from the CI run - it seems like it didn't upload a coverage report: 
```
Coverage report will not be uploaded for this build.
```

Reply:
Might be a permissions issue, the only tabs I see are Test Summary, Configuration, Build Timing and Build Parameters.

Regarding testing the function: it's a bit tricky since all it does is create a new object. Does encapsulating the codec client creation in a factory and asserting on the params passed to the factory seem like a reasonable way to test this? 

I was also considering just instantiating a ProdHttpHealthChecker and calling createCodecClient, but then it's tricky to ensure that the right kind of codec client is created without adding new functions to `Http::CodecClient`.

Reply:
Test coverage is showing that 2 out of 4 branches are covered, so at least there's some coverage. Happy to add more if you think it'd be helpful

Reply:
If you can figure out how to add it in a non-stupid way I would go for it. If not, don't worry about it. Can you fix the doc nit above though?

Reply:
Took a stab at it. Ended up a little bit gross (is there a way to not be strict about mock interactions?) but lmk what you think

Reply:
yeah, use `NiceMock` :) 

Can you switch that in and check the coverage test failure?

Reply:
`NiceMock` did it! the coverage failure was due to a missing `EXPECT_CALL` so using `NiceMock` fixed it